browser/src/Editor/NeovimEditor.tsx

```diff
@@ -283,6 +283,30 @@ export class NeovimEditor implements IEditor {
             }
         })
 
+        const normalizePath = (fileName: string) => fileName.split("\\").join("/")
+
+        ipcRenderer.on("open-files", (_evt: any, message: string, files: string[]) => {
+
+            // Open the first file.
+            // If the current buffer is named, then open in a new tab or split.
+            // If the current buffer has modifications pending, then open in a new tab or split.
+            // If the current buffer had no modification and is a new file, open in the current buffer.
+
+            this._neovimInstance.command("if bufname('%') != ''\n" +
+                        "    exec \"" + message + normalizePath(files[0]) + "\"\n" +
+                        "elseif &modified\n" +
+                        "    exec \"" + message + normalizePath(files[0]) + "\"\n" +
+                        "else\n" +
+                        "    exec \":e " + normalizePath(files[0]) + "\"\n" + 
+                        "endif")
+
+            // Open any subsequent files in a tab or split.
+            for (let i = 1; i < files.length; i++) {
```

Comment:
As you mentioned, we'd want to chain the promises. I believe Neovim will execute these in-order, but it's still safer to wait for each command to complete. Chaining the promises in a loop gets a little messy, but we could leverage async/await potentially:
```
         const openFiles = async (message: string, files: string[]) => {

            // Open the first file.
            // If the current buffer is named, then open in a new tab or split.
            // If the current buffer has modifications pending, then open in a new tab or split.
            // If the current buffer had no modification and is a new file, open in the current buffer.
            await this._neovimInstance.command("if bufname('%') != ''\n" +
                        "    exec \"" + message + normalizePath(files[0]) + "\"\n" +
                        "elseif &modified\n" +
                        "    exec \"" + message + normalizePath(files[0]) + "\"\n" +
                        "else\n" +
                        "    exec \":e " + normalizePath(files[0]) + "\"\n" + 
                        "endif")

            // Open any subsequent files in a tab or split.
            for (let i = 1; i < files.length; i++) {
                await this._neovimInstance.command("exec \"" + message + " " + normalizePath(files[i]) + "\"")
            }
        }

        ipcRenderer.on("open-files", (_evt: any, message: string, files: string[]) => {
            openFiles(message, files)
        })
```

There's still a hanging promise at the end of the chain, and ideally we'd queue things up in a more disciplined way (nothing stops another area of the code from calling command!). This could even be handling internally by Neovim instance, by queuing or chaining the commands.